### PR TITLE
adding --disable-pruning switch

### PR DESCRIPTION
**Pull request by unclear0122**
Added `--disable-pruning` flag to switch off block pruning if required.

Default value: false -> pruning enabled.

Run `./rosetta-ravencoin --disable-pruning=true` to disable pruning thread.

### CODE REVIEW

**Comment by unclear0122** [comment]
Added ravenutil (partial fork of btcutil). 
First attempt to integrate construction API

File: ravencoin/chaincfg/params.go [hunk 1]
```diff
@@ -411,8 +384,8 @@ var MainNetParams = Params{
 	Bech32HRPSegwit: "rc", // always bc for main net
 
 	// Address encoding magics
-	PubKeyHashAddrID:        0x00, // starts with 1
-	ScriptHashAddrID:        0x05, // starts with 3
+	PubKeyHashAddrID:        0x3c, // starts with R
```
Comment (ben-abraham) [inline ravencoin/chaincfg/params.go:387]: Good catch 👍

File: ravencoin/txscript/data/script_tests.json [hunk 2]
```diff
@@ -0,0 +1,1882 @@
+[
```
Comment (ben-abraham) [inline ravencoin/txscript/data/script_tests.json:1]: at *some* point we should probably update this file, but I would call it low-priority as this isn't related to the rosetta testing, just a test imported from the btcd repo.

File: ravencoin/txscript/opcode.go [hunk 3]
```diff
@@ -0,0 +1,2454 @@
...
+	OP_NOP8                = 0xb7 // 183
+	OP_NOP9                = 0xb8 // 184
+	OP_NOP10               = 0xb9 // 185
+	OP_UNKNOWN186          = 0xba // 186
+	OP_UNKNOWN187          = 0xbb // 187
+	OP_UNKNOWN188          = 0xbc // 188
+	OP_UNKNOWN189          = 0xbd // 189
+	OP_UNKNOWN190          = 0xbe // 190
+	OP_UNKNOWN191          = 0xbf // 191
+	OP_UNKNOWN192          = 0xc0 // 192
```
Comment (ben-abraham) [inline ravencoin/txscript/opcode.go:234]: Need to add OP_RVN_ASSET opcode at 0xC0 and update code below.

**Pull request merged**